### ISSUE TITLE
Variables not detected by linter

### ISSUE THREAD

**Issue by justinethier** [joereynolds/silver-doodle#9]
Variables are not always found by the linter:

```
(define TEST 1) ; OK               
(let TEST 1) ; OK, but bad syntax
;; None of these variables are detected:
(define (TEST) 1)
((lambda (TEST TEST2) 1) 1 2)                     
(let ((TEST 1)) TEST)
```

**Comment by joereynolds** [comment]
Interesting, thank's for the heads up, I'll see what I can do!
- [x] Fix the bug
- [x] Write test cases with the above tests

**Comment by joereynolds** [comment]
Hey @justinethier ,

This has been fixed but has also made me aware of another bug (which is good...?).
It will incorrectly lint a name in a string.

i.e.

`(define x "throws_an_error")`

Will throw an error :(
I've made an issue for this and thanks for helping!

Here's the fix 2e2972d76037533b5965bd6e8c793ccdf37a2c63
